extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -571,7 +572,11 @@ public TaskStatus runTask(final Closer closer)
       );
       context.writeReports(
           id(),
-          TaskReport.buildTaskReports(new MSQTaskReport(id(), taskReportPayload))
+          TaskReport.buildTaskReports(new MSQTaskReport(
+              id(),
+              task.getContextValue(DruidMetrics.TAGS),
```

Comment:
I guess you can put tags as well here. But the correct location should be 
`MSQTaskReportPayload`

Reply:
Since the tags would be in the context, and the report already has the task context so I think no change should be required here no ?

Reply:
But it would be already part of context so this should not be required.

Reply:
The tags can contain some specific info about the task such as `msq_async_download` can be a tag, it will be generated from code based on task spec not provided by user through context.
Tags are also included in task report as external systems can read task report and rely on the tags to know the purpose of the task and take actions based on it.

Reply:
> Since the tags would be in the context, and the report already has the task context so I think no change should be required here no ?

I think you mean `task query context` field, it's different from `Task.context` which all task types can have on the top level.